main.js

```diff
@@ -62,6 +63,20 @@ process.on("SIGUSR2", () => {
 });
 
 crawler = new Crawler();
-crawler.run();
 
+const app = express();
+const port = 3000;
+console.log("crawler created");
+app.get("/", (req, res) => {
+  if (crawler) {
+    res.send("Crawl initiated!");
+    crawler.run();
```

Comment:
Every time a client does a GET to `/` it will start a crawl? I believe that the crawler will call `process.exit()` once the crawl completes, which will stop your service?